R/rows.R

```diff
@@ -0,0 +1,610 @@
+#' @export
+#' @inheritParams dplyr::rows_insert
+#' @param conflict How should keys in `y` that already exist in `x` be handled?
+#'   One of:
+#'   - `"error"`, the default, will error if there are any keys in `y` that
+#'     already exist in `x`. This requires a corresponding unique constraint in
+#'     `x`.
+#'   - `"ignore"` will not insert rows in `y` with keys that already exist in `y`.
+#' @param returning Columns to return.
+#'
+#' @importFrom dplyr rows_insert
+#' @rdname rows-db
+rows_insert.tbl_lazy <- function(x,
+                                 y,
+                                 by = NULL,
+                                 ...,
+                                 conflict = c("error", "ignore"),
+                                 copy = FALSE,
+                                 in_place = FALSE,
+                                 returning = NULL) {
+  check_dots_empty()
+  rows_check_in_place(x, in_place)
+  name <- target_table_name(x, in_place)
+
+  conflict <- rows_check_conflict(conflict)
+
+  y <- auto_copy(x, y, copy = copy)
+
+  rows_check_containment(x, y)
+
+  by <- rows_check_by(by, y)
+
+  rows_check_key(x, by, "x")
+  rows_check_key(y, by, "y", unique = TRUE)
+
+  new_columns <- setdiff(colnames(y), by)
+
+  returning_cols <- rows_check_returning(x, returning, enexpr(returning))
+
+  if (!is_null(name)) {
+    if (conflict == "error") {
```

Comment:
I think it should be possible to mute this message. But how? Add another argument `quiet`? And should this inform how to add an index via `db_create_index()`?

Reply:
It would also be possible to add an option `check` that collects the data and then checks this but I'm not really sure this is worth it as the unique index makes much more sense.

Reply:
`conflict = "allow"` or `conflict = "silent"` ?

Reply:
I've proposed to change this to `conflict = "drop"` in the dplyr PR, and have `conflict = "ignore"` allow duplicate row. Hope we can reach a consensus.